kf_lib_data_ingest/etl/ingest_pipeline.py

```diff
@@ -192,11 +223,24 @@ def run(self):
         try:
             # Iterate over stages and execute them
             output = None
+            previous_stage = None
             for stage in self._iterate_stages():
+                # Only run stages that were specified in stages_to_run
+                if stage.stage_type not in self.stages_to_run:
+                    previous_stage = stage
+                    continue
+
                 self.stages[stage.stage_type] = stage
+
                 if isinstance(stage, ExtractStage):
                     output = stage.run()  # First stage gets no input
                 else:
+                    if previous_stage:
+                        self.logger.info(
+                            'Loading previously cached output from '
+                            f'{type(previous_stage).__name__}'
+                        )
+                        output = previous_stage.read_output()
```

Comment:
you need to set previous_stage = None here, otherwise if you set stages='tl' the load stage will be fed extract output.